packages/react/src/primitives/PasswordField/__tests__/ShowPasswordButton.test.tsx

```diff
@@ -21,33 +21,55 @@ describe('ShowPasswordButton component', () => {
     const ref = React.createRef<HTMLButtonElement>();
     render(<ShowPasswordButton fieldType="password" ref={ref} />);
 
-    await screen.findByRole('button');
+    await screen.findByRole('switch');
 
-    expect(ref.current.nodeName).toBe('BUTTON');
+    expect(ref.current?.nodeName).toBe('BUTTON');
   });
 
-  it('should set correct ariaLabel for fieldType password', async () => {
-    const ref = React.createRef<HTMLButtonElement>();
-    render(<ShowPasswordButton fieldType="password" ref={ref} />);
+  it('should toggle field type, screen reader context, and aria-checked when clicked', async () => {
```

Comment:
Nice!

Out of curiosity, I just searched if there's any a11y test api from react testing library, and I found this https://testing-library.com/docs/dom-testing-library/api-accessibility/. Maybe you have already been aware of it. I wonder if there's anything we can use?